crispr_sites/crispr_sites.cpp

```diff
@@ -338,30 +343,51 @@ void scan_stdin() {
     // and then merging incrementally with c++ algorithm set_union,
     // rather than doing a huge sort at the end.   Parallelizing, esp on GPU,
     // could yield phenomenal speedup if we ever need to run this program fast.
+
+    // Counting the multiplicity of guides is used for DASH guide
+    // creation in dashdat -dynerman
+    
     cerr << "Sorting " << results.size() << " candidate guides." << endl;
     sort(results.begin(), results.end());
 
     // 0 is not a valid code
     int64_t last = 0;
+    int32_t count = 0;
     for (auto it = results.begin();  it != results.end();  ++it) {
         if (*it != last) {
             ++guides;
+
+            results_unique.push_back(make_tuple(*it, count));
```

Comment:
Are forgetting to do this at the end of the loop?

Reply:
You mean you're worried about another off-by 1 error? I think it's OK - every time the loop encounters a new code it'll add the new code to the unique list.

I diff'd the output of the old style loop and the new one and they matched, at least on the input mouse fasta I'm playing with.

Reply:
Also can you make this conditional on `if (output_counts)`?   Otherwise you will be doubling the memory footprint for the human genome.   In fact, could you run this on the human genome as outlined in the README just to confirm it still works? :)

Reply:
I mean how do you get a correct count for the last unique 20-mer?

Reply:
Your counts are offset by 1, and you are missing the count for the last unique 20-mer.